### DISCUSSION TITLE
Cannot find file specified

### DISCUSSION THREAD

**Discussion by Jillinger** [osm2pgsql-dev/osm2pgsql#1604] [Q&A] [answered]
What could be wrong?
I ran...
`osm2pgsql -c -d myDatabase -U postgres -H localhost -S default.style -r pbf inputfile.osm.pbf`

I get this...
```
2022-01-03 19:55:03  Database version: 14.1
2022-01-03 19:55:03  PostGIS version: 3.1
2022-01-03 19:55:03  Setting up table 'planet_osm_point'
2022-01-03 19:55:04  Setting up table 'planet_osm_line'
2022-01-03 19:55:04  Setting up table 'planet_osm_polygon'
2022-01-03 19:55:05  Setting up table 'planet_osm_roads'
2022-01-03 19:55:05  ERROR: Open failed for 'ohdearme': The system cannot
find the file specified.
```
Am I missing something?

**Comment by lonvia** [comment, marked as answer]
My guess is that you have removed the '-W ohdearme' from the command line before posting? '-W' does not take the password as a parameter. It just forces a prompt for the password to appear. If you want to use osm2pgsql in a script and need a password for your database, then a password file is the recommended way to do that. See the [Postgresql documentation](https://www.postgresql.org/docs/current/libpq-pgpass.html) for information on its format.

PS: I've deleted the old revision of the question now but it's been out there for a while.

**Reply by Jillinger** [reply]
Looks like removing the password worked. Thanks.
This is a different topic. I have an error which I don't understand, since I know nothing about the input file, except that I downloaded it. 
_****_Edit_**. I forgot. I think this is a file I converted using JOSM... I think... or was it ogr2ogr... I don't remember. Is this something I can edit with software?**_
Can you point me to where I can get info to help me understand what to do with this: `Negative OSM object ids are not allowed`
The only thing I found was [this](https://github.com/openstreetmap/osm2pgsql/issues/1097), which doesn't help.

**Reply by mmd-osm** [reply]
If you make changes in JOSM without uploading them to OSM, and save the result as OSM XML file, JOSM uses negative object id numbers for your changes. osm2pgsql does no longer support such negative object ids, which is what the referenced Github issue is  about.

Maybe try https://help.openstreetmap.org/ to discuss parts of your workflow which aren't directly related to osm2pgsql.

**Reply by Jillinger** [reply]
Thanks
